dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1100,4 +1103,52 @@ private SourceFile getMainDescriptorFile(WorkflowVersion workflowVersion) {
         return mainDescriptor;
     }
 
+
+    @PUT
+    @Timed
+    @UnitOfWork
+    @Path("/{workflowId}/star")
+    @ApiOperation(value = "Stars a workflow.")
+    public void starEntry(@ApiParam(hidden = true) @Auth User user,
```

Comment:
This code is almost the same as that in DockerRepoResource. 
See you can write a function that can be re-used between the two for starring and unstarring entries.